ogr/ogrsf_frmts/xodr/ogrxodrdatasource.cpp

```diff
@@ -0,0 +1,188 @@
+/******************************************************************************
+ *
+ * Project:  OpenGIS Simple Features for OpenDRIVE
+ * Purpose:  Implementation of OGRXODRDataSource.
+ * Author:   Michael Scholz, German Aerospace Center (DLR)
+ *           Gülsen Bardak, German Aerospace Center (DLR)        
+ *
+ ******************************************************************************
+ * Copyright 2024 German Aerospace Center (DLR), Institute of Transportation Systems
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#include "ogr_xodr.h"
+using namespace odr;
+using namespace pugi;
+using namespace std;
+
+bool OGRXODRDataSource::Open(const char *pszFilename, CSLConstList openOptions)
+{
+    VSILFILE *file = nullptr;
+    file = VSIFOpenL(pszFilename, "r");
+    if (file == nullptr)
+        return FALSE;
+
```

Comment:
This leaks the VSILFILE*, and is not necessary due OGRXODRDriverOpen() testing that poOpenInfo->fpL != nullptr

```suggestion
```